catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/BasicTypes.java

```diff
@@ -99,6 +101,8 @@
 
     public static final String VALIDATION_ERRORS = "validation-errors";
 
+    private static final Map<String, AttributeType> ATTRIBUTE_TYPE_MAP = new HashMap<>();
+
```

Comment:
This static initializer is terribly long and contains a lot of duplication. Could we:

1. Create a `private` inner class for `AttributeType` that takes in the `AttributeFormat` and `Class` to return instead of repeating the interface implementation every time?
2. Delegate the creation of the `AttributeType` and addition to the `ATTRIBUTE_TYPE_MAP` to a `private static` method instead of inlining everything in the initializer?

I know this is probably out of scope, but I think it'd help a lot and shouldn't be too difficult to do.